### PR TITLE
Feature is placementgroup

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -47,4 +47,4 @@ require (
 	gotest.tools v2.2.0+incompatible
 )
 
-replace github.com/softlayer/softlayer-go v1.0.3 => github.com/IBM-Cloud/softlayer-go v1.0.3-tf
+replace github.com/softlayer/softlayer-go v1.0.3 => github.com/IBM-Cloud/softlayer-go v1.0.3-tf
```
Comment (kavya498) [inline go.mod:50]: What change has been made in go.mod?

File: ibm/data_source_ibm_is_instance.go [hunk 2]
```diff
@@ -441,6 +441,54 @@ func dataSourceIBMISInstance() *schema.Resource {
 					},
 				},
 			},
+			"placement_target": &schema.Schema{
```
Comment (kavya498) [inline ibm/data_source_ibm_is_instance.go:444]: Docs is not updated for this feature in is_instance datasource

Since it is a list does it have to be placement_targets?

File: ibm/data_source_ibm_is_instance.go [hunk 3]
```diff
@@ -495,6 +543,13 @@ func instanceGetByName(d *schema.ResourceData, meta interface{}, name string) er
 			}
 			d.Set(isInstanceCPU, cpuList)
 
+			if instance.PlacementTarget != nil {
+				placementTargetMap := resourceIbmIsInstanceInstancePlacementToMap(*instance.PlacementTarget.(*vpcv1.InstancePlacementTarget))
+				if err = d.Set("placement_target", []map[string]interface{}{placementTargetMap}); err != nil {
+					return fmt.Errorf("Error setting placement_target: %s", err)
```
Comment (kavya498) [inline ibm/data_source_ibm_is_instance.go:549]: do not return error

File: ibm/data_source_ibm_is_instances.go [hunk 4]
```diff
@@ -60,6 +60,34 @@ func dataSourceIBMISInstances() *schema.Resource {
...
+			},
+
+			"dedicated_host": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"dedicatedhost_name"},
+				Description:   "ID of the dedicated host to filter the instances attached to it",
+			},
+
+			"placementgroup_name": {
```
Comment (kavya498) [inline ibm/data_source_ibm_is_instances.go:77]: rename `placementgroup_name` to `placement_group_name`

File: ibm/data_source_ibm_is_instances.go [hunk 5]
```diff
@@ -60,6 +60,34 @@ func dataSourceIBMISInstances() *schema.Resource {
 				Description: "Instance resource group",
 			},
 
+			"dedicatedhost_name": {
```
Comment (kavya498) [inline ibm/data_source_ibm_is_instances.go:63]: rename `dedicatedhost_name` to dedicated_host_name
Please update docs as well

File: ibm/data_source_ibm_is_placement_group.go [hunk 6]
```diff
@@ -0,0 +1,210 @@
+/**
```
Comment (kavya498) [inline ibm/data_source_ibm_is_placement_group.go:1]: change copy right statement to below statements in all new files
```
// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
// Licensed under the Mozilla Public License v2.0
```

File: ibm/resource_ibm_is_placement_group.go [hunk 7]
```diff
@@ -0,0 +1,467 @@
...
+	if err != nil {
+		log.Printf("[DEBUG] CreatePlacementGroupWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+
+	d.SetId(*placementGroup.ID)
+
+	_, err = isWaitForPlacementGroupAvailable(vpcClient, d.Id(), d.Timeout(schema.TimeoutCreate), d)
+	if err != nil {
+		return diag.FromErr(err)
```
Comment (kavya498) [inline ibm/resource_ibm_is_placement_group.go:204]: Please add error strings in all wait errors
`return diag.FromErr(err)` ==> `return diag.FromErr(fmt.Errorf("[ERROR] Error waiting for placement group to be available %s", err)`

File: ibm/data_source_ibm_is_instance.go (hunk 2 above)
Comment (deepaksibm) [reply ibm/data_source_ibm_is_instance.go:444]: It will have only one element. There is only one placement target. List is for having its attributes like CRN, Id etc.

File: ibm/resource_ibm_is_instance_template.go [hunk 8]
```diff
@@ -292,54 +313,38 @@ func resourceIBMISInstanceTemplate() *schema.Resource {
...
 				Type:        schema.TypeList,
 				Computed:    true,
-				Description: "The placement restrictions to use for the virtual server instance.",
+				Description: "The placement restrictions for the virtual server instance. For the target tobe changed, the instance `status` must be `stopping` or `stopped`.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"id": {
+						"id": &schema.Schema{
 							Type:        schema.TypeString,
-							Computed:    true,
```
Comment (hkantare) [inline ibm/resource_ibm_is_instance_template.go:319]: Why is this changed to optional from computed?
isInstanceTemplatePlacementTarget is a computed variable

Comment (hkantare) [reply ibm/resource_ibm_is_instance_template.go:319]: similarly other
